tests/test_boundless_read.py

```diff
@@ -0,0 +1,19 @@
+"""Test of boundless reads"""
+
+from hypothesis import given
+import hypothesis.strategies as st
+import numpy
+
+import rasterio
+from rasterio.windows import Window
+
+
+@given(st.integers(min_value=0, max_value=700))
+def test_outer_boundless_pixel_fidelity(size):
+    """An outer boundless read doesn't change pixels"""
+    path_rgb_byte_tif = 'tests/data/RGB.byte.tif'
+    with rasterio.open(path_rgb_byte_tif) as dataset:
+        rgb = dataset.read()
+        rgb_padded = dataset.read(window=Window(-size, -size, dataset.width + 2 * size, dataset.height + 2 * size), boundless=True)
```

Comment:
Declare variables for `padded_width = dataset.width + 2 * w_padding` and likewise for height, and use here and in the assertion.